File: src/main/java/com/wxl/dyttcrawler/core/CrawlerListener.java
package com.wxl.dyttcrawler.core;

import us.codecraft.webmagic.Request;
import us.codecraft.webmagic.Task;

/**
 * Create by wuxingle on 2020/7/19
 * 请求处理监听
 */
public interface CrawlerListener {

    /**
     * 成功
     */
    void onSuccess(Request request, Task task);

    /**
     * 失败
     */
    void onError(Request request, Task task);
}

File: src/main/kotlin/com/wxl/dyttcrawler/core/CrawlerListener.kt
package com.wxl.dyttcrawler.core

import us.codecraft.webmagic.Request
import us.codecraft.webmagic.Task

/**
 * Create by wuxingle on 2021/10/08
 * 请求处理监听
 */
interface CrawlerListener {

    /**
     * 成功处理
     */
    fun onSuccess(request: Request, task: Task)

    /**
     * 失败处理
     */
    fun onError(request: Request, task: Task, e: Exception?)
}
